### PR TITLE
Corrections to Queue code.

### PR DESCRIPTION
**Pull request by tomsons26**
We were using wrong functions in the current code we have that uses the queues.

Operators [] were removed cause the only legal way to fetch in this class is via Fetch_Head and Fetch_From_Head. Operator = was removed cause there should never be a need to do this because these are static buffers.
Tail functions don't show in any binary we have and AFAIK fetching the tail and removing the tail would just crash this since it is the tail - the end of the queue, there's nothing there.

### CODE REVIEW

File: src/game/engine/queue.cpp [hunk 1]
```diff
@@ -68,7 +71,76 @@ void Queue_Playback()
     void (*func)() = reinterpret_cast<void (*)()>(0x0052BDEC);
     return func();
 #else
-    // TODO
+    static int _mx, _my;
```
Comment (OmniBlade) [inline src/game/engine/queue.cpp]: Declare these on separate lines please, we don't do multiple variable declarations on the same line for Chronoshift code style.

File: src/game/engine/queue.cpp [hunk 2]
```diff
@@ -68,7 +71,76 @@ void Queue_Playback()
...
+
+    GameEventClass event;
```
Comment (OmniBlade) [inline src/game/engine/queue.cpp]: New line please.

File: src/game/engine/queue.cpp [hunk 3]
```diff
@@ -68,7 +71,76 @@ void Queue_Playback()
...
+                    captainslog_debug("Queue_Playback() - Reading event count failed");
+                    read = false;
+                } else {
+                    for (int i = 0; i < numevents; ++i) {
+                        if (g_Session.Recording_File().Read(&event, sizeof(GameEventClass)) != sizeof(GameEventClass)) {
+                            captainslog_debug("Queue_Playback() - Reading events failed");
+                            read = false;
+                            break;
+                        }
+                        event.Set_Executed(false); // i think...
```
Comment (OmniBlade) [inline src/game/engine/queue.cpp]: New line please.

File: src/game/engine/queue.cpp [hunk 4]
```diff
@@ -68,7 +71,76 @@ void Queue_Playback()
...
+                        g_ScheduledEvents.Add(event);
+                    }
+                }
+
+                if (read == false) {
+                    g_GameActive = 0;
+                } else {
+                    int players = 0;
+                    HousesType house;
+                    if (g_Session.Game_To_Play() == GAME_CAMPAIGN) {
```
Comment (OmniBlade) [inline src/game/engine/queue.cpp]: New line please

File: src/game/engine/queue.cpp [hunk 5]
```diff
@@ -140,36 +214,70 @@ BOOL Queue_Mission(TargetClass whom, MissionType mission, target_t target, targe
...
+BOOL Queue_Mission_Formation(
+    TargetClass whom, MissionType mission, target_t target, target_t dest, SpeedType form_speed, MPHType form_max_speed)
 {
     GameEventClass ev(whom, mission, target, dest, form_speed, form_max_speed);
     return g_OutgoingEvents.Add(ev);
 }
 
-BOOL Execute_ScheduledEvents(int house_count, HousesType houses, ConnectionManagerClass *conn_mgr, TCountDownTimerClass<FrameTimerClass> *a4, signed int *a5, unsigned short *a6, unsigned short *a7)
+BOOL Execute_Scheduled_Events(int house_count, HousesType houses, ConnectionManagerClass *conn_mgr,
+    TCountDownTimerClass<FrameTimerClass> *a4, signed int *a5, unsigned short *a6, unsigned short *a7)
```
Comment (OmniBlade) [inline src/game/engine/queue.cpp]: signed int should be just int and unsigned short should be uint16_t.

File: src/game/engine/queue.cpp [hunk 6]
```diff
@@ -140,36 +214,70 @@ BOOL Queue_Mission(TargetClass whom, MissionType mission, target_t target, targe
...
-            if (ev.Get_Event_Frame() >= g_GameFrame) {
-                break;
-            }
+        g_Keyboard->Check();
+        if ( conn_mgr != nullptr ) {
+            //Update_Queue_Mono(conn_mgr, 7);
         }
-        g_ScheduledEvents.Remove_Tail();
+
+        GameEventClass &ev = g_ScheduledEvents.Fetch_Head();
```
Comment (OmniBlade) [inline src/game/engine/queue.cpp:179]: More new lines needed around here.

**Review by OmniBlade — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]

**Review by OmniBlade — APPROVED** [review]

**Pull request merged**